### PR TITLE
Adding steps to follow when handling API/ABI breakage of a package

### LINKED ISSUE #1581: [DOC] Explain how to handle a ABI/API breakage of a package
Currently when a package breaks, there are a few steps people need to go through. Roughly they are:

* Rebuild the package with corrected `run_exports` (build number bump)
* Hot-fix repodata of previous package to apply the right `run_exports` 
* Hot-fix the repodata of dependencies to include corrected pinnings for the package

There may be other things here I'm forgetting (feel free to just edit the OP or comment below). Would be good to have this documented better so we can just point people to the appropriate doc page when an issue like this arises

### PR DESCRIPTION
**Pull request by ssurbhi560**
PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

fixes #1581

### CODE REVIEW

**Comment by ssurbhi560** [comment]
@kathatherine @jakirkham, please review this and let me know if I should add/remove anything.

File: src/user/faq.rst [hunk 1]
```diff
@@ -80,3 +80,20 @@ FAQ
...
   
   If you need to compile CUDA code, even if it involves only CUDA host APIs, you will still need a valid CUDA Toolkit installed locally and use it. Please refer to `NVCC's documentation <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`_ for the CUDA compiler usage and `CUDA Programming Guide <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html>`_ for general CUDA programming.
+
+.. _faq_api_abi_breakage:
+
+:ref:`(Q) <faq_api_abi_breakage>` **How to handle a ABI/API breakage of a package?**
```
[the remark this answers is not in the document]
Comment (ssurbhi560) [reply src/user/faq.rst:86]: I am really sorry @isuruf, I can't seem to understand your question. Are you suggesting that we should also add in brief what could be the possible reasons for API/ABI breakage of a package? I am new here so, I am not very well versed with everything and a little confused as well, but I am learning, so I would be grateful if you could please help.

[the remark this answers is not in the document]
Comment (isuruf) [reply src/user/faq.rst]: I mean is it a breakage in a version number (if so, is it covered by `run_exports`) or is it a breakage in a build number?

[the remark this answers is not in the document]
Comment (jakirkham) [reply src/user/faq.rst]: IIUC Surbhi is one of the Outreachy applicants so may not know relevant technical details (though I could be wrong about this)

If there is particular verbiage or structuring of this, would suggest we just propose those changes

**Comment by jakirkham** [comment]
cc @beckermr (for awareness)

File: src/user/faq.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (ssurbhi560) [reply src/user/faq.rst]: @jakirkham, that's right, I am an outreachy applicant. I am still learning and trying to understand how everything works, and hope to be able to help more in future, but till then I would be glad if you folks could help me with things.

[the remark this answers is not in the document]
Comment (beckermr) [reply src/user/faq.rst]: This faq is supposed to cover what happens when we ship a package with a run export at say x.x, but we should have had that be say x.x.x. Typically this happens when there is an ABI break during a patch or minor version update. I think the corrected title below is better. 

We could also simply say how to fix incorrect run exports or something along those lines.

[the remark this answers is not in the document]
Comment (ssurbhi560) [reply src/user/faq.rst]: Ah, okay! Thanks a lot, @beckermr, for explaining this. So, should I go with the same title for the FAQ or change it to "How to fix `run_exports` to handle ABI breakage?" as also suggested by Matt, @jakirkham @isuruf, what do you folks say?

[the remark this answers is not in the document]
Comment (beckermr) [reply src/user/faq.rst]: Let's go with the title as is for now.

File: src/user/faq.rst [hunk 2]
```diff
@@ -80,3 +80,20 @@ FAQ
...
   If you need to compile CUDA code, even if it involves only CUDA host APIs, you will still need a valid CUDA Toolkit installed locally and use it. Please refer to `NVCC's documentation <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`_ for the CUDA compiler usage and `CUDA Programming Guide <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html>`_ for general CUDA programming.
+
+.. _faq_api_abi_breakage:
+
+:ref:`(Q) <faq_api_abi_breakage>` **How to handle breaking of a package due to ABI/API incompatibility?**
+
+  If your package breaks due to ABI/API incompatibility, here are a few steps you can take to fix it:
+
+  - Rebuild the package with corrected ``run_exports``.
+  - Hot-fix repodata of previous package to apply the right ``run_exports``.
```
Comment (isuruf) [inline src/user/faq.rst:91]: `run_exports` cannot be hot-fixed

File: src/user/faq.rst [hunk 3]
```diff
@@ -80,3 +80,20 @@ FAQ
...
   
   If you need to compile CUDA code, even if it involves only CUDA host APIs, you will still need a valid CUDA Toolkit installed locally and use it. Please refer to `NVCC's documentation <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`_ for the CUDA compiler usage and `CUDA Programming Guide <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html>`_ for general CUDA programming.
+
+.. _faq_api_abi_breakage:
+
+:ref:`(Q) <faq_api_abi_breakage>` **How to handle breaking of a package due to ABI/API incompatibility?**
```
Comment (isuruf) [inline src/user/faq.rst:86]:
```suggestion
:ref:`(Q) <faq_api_abi_breakage>` **How to handle breaking of a package due to ABI incompatibility?**
```

File: src/user/faq.rst [hunk 4]
```diff
@@ -80,3 +80,20 @@ FAQ
...
+
+  If your package breaks due to ABI/API incompatibility, here are a few steps you can take to fix it:
```
Comment (isuruf) [inline src/user/faq.rst:88]:
```suggestion
  If your package breaks ABI with a version bump, here are a few steps you can take to fix it:
```

File: src/user/faq.rst [hunk 5]
```diff
@@ -80,3 +80,19 @@ FAQ
...
   
   If you need to compile CUDA code, even if it involves only CUDA host APIs, you will still need a valid CUDA Toolkit installed locally and use it. Please refer to `NVCC's documentation <https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html>`_ for the CUDA compiler usage and `CUDA Programming Guide <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html>`_ for general CUDA programming.
+
+.. _faq_abi_incompatibility:
+
+:ref:`(Q) <faq_abi_incompatibility>` **How to handle breaking of a package due to ABI incompatibility?**
+
+  If your package breaks ABI with a version bump, here are a few steps you can take to fix it:
+
+  - Rebuild the package with corrected ``run_exports``.
```
Comment (isuruf) [inline src/user/faq.rst:90]:
```suggestion
  - Rebuild the new version with corrected ``run_exports``.
  - Rebuild the old version with corrected ``run_exports``.
```

**Review by kathatherine — APPROVED** [review]
This looks good to me if it looks good to Core. Thanks!

**Comment by jakirkham** [comment]
Anything else needed here @conda-forge/core ?

**Comment by hmaarrfk** [comment]
A PR should be made to the migration feedstock adding the old pinning as the current pinning and the new pinning in a migrator.

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
@hmaarrfk could you please modify the text above as you see fit? Either using a suggestion or committing the change?

File: src/user/faq.rst [hunk 6]
```diff
@@ -80,3 +80,20 @@ FAQ
...
+
+.. _faq_abi_incompatibility:
+
+:ref:`(Q) <faq_abi_incompatibility>` **How to handle breaking of a package due to ABI incompatibility?**
+
+  If your package breaks ABI with a version bump, here are a few steps you can take to fix it: 
+
+  - Rebuild the new version with corrected ``run_exports``.
+  - Rebuild the old version with corrected ``run_exports``.
+  - Hot-fix the repodata of dependencies to include corrected pinnings for the package.
```
Comment (jakirkham) [inline src/user/faq.rst:92]:
```suggestion
  - Hot-fix the repodata of dependencies to include corrected pinnings for the package.
  - Add a PR to pin the old version in ``conda-forge-pinning`` (if not already present)
  - Open a migrator following `CFEP-09 <https://github.com/conda-forge/cfep/blob/main/cfep-09.md>`_
```

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! :pray:

Understand these things can always be improved (and folks should feel free to adjust the wording in follow up PRs). Having at least some info (as opposed to none) is incredibly useful for people trying to figure out what to do. We will likely continue to improve on this by having folks use these instructions and update them based on anything that comes up

**Pull request merged**